src/sentry/static/sentry/app/components/events/interfaces/stacktraceLinkModal.tsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import Modal from 'react-bootstrap/lib/Modal';
+import styled from '@emotion/styled';
+
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import AsyncComponent from 'app/components/asyncComponent';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {t, tct} from 'app/locale';
+import space from 'app/styles/space';
+import {Integration, Organization, Project} from 'app/types';
+import {getIntegrationIcon} from 'app/utils/integrationUtil';
+import InputField from 'app/views/settings/components/forms/inputField';
+
+import {OpenInContainer} from './openInContextLine';
+
+type Props = AsyncComponent['props'] & {
+  filename: string;
+  organization: Organization;
+  project: Project;
+  integrations: Integration[];
+  onClose: () => void;
+};
+
+type State = AsyncComponent['state'] & {
+  showModal: boolean;
+  sourceCodeInput: string;
+};
+
+class StacktraceLinkModal extends AsyncComponent<Props, State> {
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      showModal: false,
+      sourceCodeInput: '',
+    };
+  }
+  openModal() {
+    this.setState({
+      showModal: true,
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      showModal: false,
+      sourceCodeInput: '',
+    });
+  }
+
+  createConfig = async configData => {
+    const {organization, project} = this.props;
+    const endpoint = `/organizations/${organization.slug}/integrations/${configData.integrationId}/repo-project-path-configs/`;
+    try {
+      await this.api.requestPromise(endpoint, {
+        method: 'POST',
+        data: {...configData, projectId: project.id},
+      });
+      addSuccessMessage(t('Stack trace link config created.'));
+    } catch (err) {
+      const errors = err?.responseJSON
+        ? Array.isArray(err?.responseJSON)
+          ? err?.responseJSON
+          : Object.values(err?.responseJSON)
+        : [];
+      const apiErrors = errors.length > 0 ? `: ${errors.join(', ')}` : '';
+      addErrorMessage(t('Unable to save configuration%s', apiErrors));
+    }
+    this.closeModal();
+    this.props.onClose();
```

Comment:
@MeredithAnya should we close the modal on failure as well?

Reply:
@scefali hmmm are you suggesting that if we fail on trying to save the configuration (even if parsing endpoint was successful) that we should keep the modal open and maybe indicate that they should try the manual set up?

Reply:
@MeredithAnya yea, something like that might work